regrid/regrid.py

```diff
@@ -0,0 +1,251 @@
+        nlat: int,
+        nlon: int,
+        interp_method: str = "bilinear",
+        config_filename: str = "config-regrid.yaml",
+    ):
+        super(Regrid, self).__init__()
+        name = self.__class__.__name__
+
+        # read configuration from yaml file
+        with open(config_filename, "r") as f:
+            contents = yaml.safe_load(f)
+            self.config = contents[name]
+
+        # specify an output resolution
+        self.nlon_o = nlon
+        self.nlat_o = nlat
+        self.interp_method = interp_method
+
+    def compute_gaussian_grid(self, nlat, nlon):
+        """Compute gaussian grid latitudes and longitudes"""
+        latitudes, _ = gaussian_latitudes(nlat // 2)
+        longitudes = np.linspace(0, 360, nlon, endpoint=False)
+        return latitudes, longitudes
+
+    def compute_latlon_grid(self, nlat, nlon):
+        """Compute regular latlong grid coordinates"""
+        latitudes = np.linspace(-90, 90, nlat, endpoint=False)
+        longitudes = np.linspace(0, 360, nlon, endpoint=False)
+        return latitudes, longitudes
+
+    def create_regridder(self, file, compute_grid=None):
+        """Create regridding instances"""
+        if compute_grid is None:
+            compute_grid = self.compute_gaussian_grid
+        self.input_path = self.config["input_path"]
+        self.rotation_file = self.config["rotation_file"]
+        self.output_path = self.config["output_path"]
+
+        # open input dataset with first file
+        ds_in = xr.open_dataset(file)
+        nlon_i = ds_in.sizes["yh"]
+        nlat_i = ds_in.sizes["xh"]
+        self.ires = f"{nlon_i}x{nlat_i}"
+        self.ores = f"{self.nlon_o}x{self.nlat_o}"
+        ds_in_t = ds_in.rename({"xh": "lon", "yh": "lat"})
+        ds_in_u = ds_in.rename({"xq": "lon", "yh": "lat"})
+        ds_in_v = ds_in.rename({"xh": "lon", "yq": "lat"})
+
+        # open rotation dataset
+        file_rot = f"{self.rotation_file}"
+        ds_rot = xr.open_dataset(file_rot)
+        ds_rot = ds_rot[["cos_rot", "sin_rot"]]
+        self.ds_rot = ds_rot.rename({"xh": "lon", "yh": "lat"})
+
+        # create output dataset
+        self.lat1d, self.lon1d = compute_grid(self.nlat_o, self.nlon_o)
+        lons, lats = np.meshgrid(self.lon1d, self.lat1d)
+        da_out_lons = xr.DataArray(lons, dims=["nx", "ny"])
+        da_out_lats = xr.DataArray(lats, dims=["nx", "ny"])
+        ds_out_lons = da_out_lons.to_dataset(name="lon")
+        ds_out_lats = da_out_lats.to_dataset(name="lat")
+        grid_out = xr.merge([ds_out_lons, ds_out_lats])
+
+        # interpolation weights files
+        wgtsfile_t_to_t = (
+            f"{self.output_path}/weights-{self.ires}.Ct.{self.ores}.Ct.bilinear.nc"
+        )
+        wgtsfile_u_to_t = (
+            f"{self.output_path}/weights-{self.ires}.Cu.{self.ires}.Ct.bilinear.nc"
+        )
+        wgtsfile_v_to_t = (
+            f"{self.output_path}/weights-{self.ires}.Cv.{self.ires}.Ct.bilinear.nc"
+        )

```

Comment:
I'd prefer to not assume a path structure, but instead expect the user to provide an option in the yaml file with these names. Something like
```suggestion
        weights_file_t2t = self.config["weights_file_t2t"]
        weights_file_u2t = self.config["weights_file_u2t"]
        weights_file_v2t = self.config["weights_file_v2t"]
```

Reply:
I have added optional paths to the weights files in the config file, so that one can use pre-generated weights file.
If not specified, it will try to create a unique name from input/output resolutions and interpolation method. Not sure if I should remove this and rely entirely on the config file. If we had jinja support in the yaml file, maybe it is possible to do so. Let me know.